add monthly blogpost - January 24 (#976)
withastro/astro.build: src/content/blog/whats-new-january-2024.mdx

## Docs

New this month in both Astro and Starlight docs: dedicated community resource pages. Check out the [community-produced educational content about Astro](https://docs.astro.build/en/community-resources/content/), and the growing list of [Starlight content](https://starlight.astro.build/resources/community-content/). Catch up on some of the classic [talks, interviews, and streams](https://docs.astro.build/en/community-resources/talks/) you might have missed.

Our weekly Discord call "Talking and Doc'ing" is a chance for you to watch Team Docs work in public, jump in a shared Codespace or Gitpod Workspace, and contribute to the docs live, together. This month's topics included:

- **Building a new `<ReadMore />` component.** After watching Chris build, everyone on the call then jumped in the code base to help find and replace all occurrences of our old "emoji + See more at this link" pattern.
- **Taking [Lunaria](https://lunaria.dev/), and its documentation, for a test drive.** We started with a new Starlight project, and finished with a site with full translation status tracking!

> - **File/URL structure organization.** Soon, the URLs for all our pages will actually correspond to the section in the sidebar in which they live. How did we not break anything with 12 people in the codebase, moving files around and updating links all at once? 🫣

Reviewer: ```suggestion
- **File/URL structure organization.** Soon, the URLs for all our pages will actually correspond to their sidebar section! How did we not break anything with 12 people in the codebase, moving files around and updating links all at once? 🫣
```

## Updates from Mission Control

- We have a [new channel on Discord](/chat) `#dev-integrations`, a new home for anyone building their own Astro integrations.
- Starlight was nominated for a Product Hunt [open-source Golden Kitty Award](https://www.producthunt.com/golden-kitty-awards/hall-of-fame?year=2023#open-source).
- We launched [The Official Astro Newsletter](/newsletter/signup). Sign up if you want to stay updated on all things happening in the Astrosphere!
- Never want to miss an Astro event? [Subscribe to our Astro Community Calendar](https://calendar.google.com/calendar/embed?src=c_g6re3u5gkl0mug3ktn1dibqhfk%40group.calendar.google.com).

## Releases

The first month of the year has already started strong with minor releases from both Astro and Starlight 💪!

- [Astro 4.1](https://astro.build/blog/astro-4/)
- [Astro 4.2](https://astro.build/blog/astro-420/) 
- [Starlight 0.16](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.16.0) 
- [Starlight 0.17](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.17.0)

## Docs

> New this month in both Astro and Starlight docs: dedicated sidebar sections with pages for community content and resources.
> 
> You'll find links to tutorials, blog posts, community recipes, videos and streams. Check out the [community-produced content about Astro](https://docs.astro.build/en/community-resources/content/), and the growing list of [Starlight content](https://starlight.astro.build/resources/community-content/). Catch up on some of the classic [talks, interviews, and streams](https://docs.astro.build/en/community-resources/talks/) you might have missed.
> 
> Open a PR with a link to your own recipes in the appropriate categories and help the Astro community add new features and functionality to their own sites!
> 
> Our weekly Discord call "Talking and Doc'ing" is a chance for you to watch Team Docs work in public, jump in a shared Codespace or Gitpod Workspace, and contribute to the docs live, together. This month's topics included:
> 
> - **Team Docs quarterly planning.** We brainstormed ideas, decided our goals, and broke down projects into sprints.
> - **Building a new `<ReadMore />` component.** Everyone on the call then jumped in the code base to help find and replace all occurrences of our old "emoji + See more at this link" pattern.
> - **Taking [Lunaria](https://lunaria.dev/), and its documentation, for a test drive.** We built a Starlight site from scratch, then everyone on the call added new languages and various pages so we would have translation statuses to track. Then we installed Lunaria and finished the call with a new i18n site, complete with translation tracking!
> - **File/URL structure organization.** Soon, the URLs for all our pages will actually correspond to the section in the sidebar in which they live. How did we not break anything with 12 people in the codebase, moving files around and updating links all at once? 🫣

Reviewer: ```suggestion
New this month in both Astro and Starlight docs: dedicated community resource pages. Check out the [community-produced educational content about Astro](https://docs.astro.build/en/community-resources/content/), and the growing list of [Starlight content](https://starlight.astro.build/resources/community-content/). Catch up on some of the classic [talks, interviews, and streams](https://docs.astro.build/en/community-resources/talks/) you might have missed.

Our weekly Discord call "Talking and Doc'ing" is a chance for you to watch Team Docs work in public, jump in a shared Codespace or Gitpod Workspace, and contribute to the docs live, together. This month's topics included:

- **Building a new `<ReadMore />` component.** After watching Chris build, everyone on the call then jumped in the code base to help find and replace all occurrences of our old "emoji + See more at this link" pattern.
- **Taking [Lunaria](https://lunaria.dev/), and its documentation, for a test drive.** We started with a new Starlight project, and finished with a site with full translation status tracking!
- **File/URL structure organization.** Soon, the URLs for all our pages will actually correspond to the section in the sidebar in which they live. How did we not break anything with 12 people in the codebase, moving files around and updating links all at once? 🫣
```

## Releases

The first month of the year has already started strong with minor releases from both Astro and Starlight 💪!

- [Astro 4.1](https://astro.build/blog/astro-4/)
- [Astro 4.2](https://astro.build/blog/astro-420/) 
- [Starlight 0.16](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.16.0) 
- [Starlight 0.17](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.17.0)

## Docs

New this month in both Astro and Starlight docs: dedicated sidebar sections with pages for community content and resources.

You'll find links to tutorials, blog posts, community recipes, videos and streams. Check out the [community-produced content about Astro](https://docs.astro.build/en/community-resources/content/), and the growing list of [Starlight content](https://starlight.astro.build/resources/community-content/). Catch up on some of the classic [talks, interviews, and streams](https://docs.astro.build/en/community-resources/talks/) you might have missed.

Open a PR with a link to your own recipes in the appropriate categories and help the Astro community add new features and functionality to their own sites!

> Our weekly Discord call "Talking and Doc'ing" is a chance for you to watch Team Docs work in public, jump in a codespace, and contribute to the docs. This month's topics included:

Reviewer: ```suggestion
Our weekly Discord call "Talking and Doc'ing" is a chance for you to watch Team Docs work in public, jump in a shared Codespace or Gitpod Workspace, and contribute to the docs live, together. This month's topics included:
```

- Team Docs quarterly planning! We brainstormed ideas, decided our goals, and broke down projects into sprints.

> - Building a new `<ReadMore />` component, and then everyone on the call jumped in the code base to help find and replace all occurrences of our old "emoji + See more at this link" pattern.

Reviewer: ```suggestion
- Building a new `<ReadMore />` component! Everyone on the call then jumped in the code base to help find and replace all occurrences of our old "emoji + See more at this link" pattern.
```

- 📺 [How to upload images with Astro and Xata](https://www.youtube.com/watch?v=e7UciRSdwYs) by @rishirajjain
- 📄 [Dark Modes](https://www.xypnox.com/blag/posts/dark-modes/) by @xypnox
- 📄 [Basic analytics with Vercel Postgres - Drizzle - Astro](https://www.thomasledoux.be/blog/basic-analytics-vercel-postgres-astro) by @tledoux
- 📺 [Page Components - better than React Server Components](https://youtu.be/XA_rS_SWQZs) by @koshchei.
- 📄 [The future of SSR is RSC](https://www.jseverywhere.dev/the-future-of-ssr-is-rsc/) by @colby.white
- 📄 [Astro + Svelte on the Internet Computer](https://blog.icacademy.at/blog/astro-svelte-ic-starter) by @rbole
- 📺 [Astro Icon just hit 1.0](https://www.youtube.com/watch?v=Tm4XMd_CL3o) by @chrispennington
- 📄 [Using HTMX with Astro Partials](https://blog.trevfox.dev/posts/htmx-and-astro-partials/) by @vxoblope
- 📄 [Sending mails with Astro and Nodemailer](https://oliverspeir.dev/garden/nodemailer/) by @oliverspeir
- 📄 [Create a web application with Astro and Drupal](https://www.linkedin.com/posts/gambinovincenzo_create-a-web-application-with-the-astrojs-activity-7156645583347236865-YLDR/) by @vincenzogambino

> - 📄 [Astro API tutorial](https://www.ceos3c.com/astro/astro-api-tutorial/)
> - 📄 [Astro Kwesforms with Rive](https://www.sitepoint.com/astro-kwesforms-rive/)
> - 📄 [Take a Qwik break from React with Astro](https://thenewstack.io/take-a-qwik-break-from-react-with-astro/)

Reviewer: ```suggestion
- 📄 [Astro API tutorial](https://www.ceos3c.com/astro/astro-api-tutorial/) by @stefanrows
- 📄 [Astro Kwesforms with Rive](https://www.sitepoint.com/astro-kwesforms-rive/) by @pauliescanlon
- 📄 [Take a Qwik break from React with Astro](https://thenewstack.io/take-a-qwik-break-from-react-with-astro/) by @pauliescanlon
```

---
title: "What's new in Astro - January 2024"
description: "2023 was a huge year for Astro, and 2024 is already shaping up to be even bigger. Let's dive into the updates!"
publishDate: "January 31, 2024"
authors:
  - elian
coverImage: "/src/content/blog/_images/whats-new-shared.webp"
socialImage: "/src/content/blog/_images/whats-new-january-2024/og.webp"
lang: "en"
---
import ShowcaseGrid from './_whats-new-components/ShowcaseGrid.astro';
import ThemeGrid from './_whats-new-components/ThemeGrid.astro';

2023 was a huge year for Astro, and 2024 is already shaping up to be even bigger. Let's dive into the updates!

## Releases

The first month of the year already started strong 💪!

- [Astro 4.1](https://astro.build/blog/astro-4/)
- [Astro 4.2](https://astro.build/blog/astro-420/) 
- [Starlight 0.16](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.16.0) 
- [Starlight 0.17](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.17.0)

## Docs

> Our weekly Discord call “Talking and Doc’ing” is a chance for you to watch Team Docs work in public, jump in a codespace, and contribute to the docs. This month’s topics included:

Reviewer: ```suggestion
New this month in both Astro and Starlight docs: dedicated sidebar sections with pages for community content and resources.

You'll find links to tutorials, blog posts, community recipes, videos and streams. Check out the [community-produced content about Astro](https://docs.astro.build/en/community-resources/content/), and the growing list of [Starlight content](https://starlight.astro.build/resources/community-content/). Catch up on some of the classic [talks, interviews, and streams](https://docs.astro.build/en/community-resources/talks/) you might have missed.

Open a PR with a link to your own recipes in the appropriate categories and help the Astro community add new features and functionality to their own sites!

Our weekly Discord call “Talking and Doc’ing” is a chance for you to watch Team Docs work in public, jump in a codespace, and contribute to the docs. This month’s topics included:
```
